synapse/rest/client/v1/login.py

```diff
@@ -284,9 +284,7 @@ async def _complete_login(
         self,
         user_id: str,
         login_submission: JsonDict,
-        callback: Optional[
-            Callable[[Dict[str, str]], Awaitable[Dict[str, str]]]
-        ] = None,
+        callback: Optional[Callable[[Dict[str, str]], Awaitable[None]]] = None,
```

Comment:
below it claims that `callback` is called after registration. It looks like it is also called after login. Might be worth fixing that up while we're here, since we're relying on that.